foo_uie_console/main.cpp

```diff
@@ -305,9 +323,19 @@ class EdgeStyleMenuNode : public uie::menu_node_popup_t {
     std::vector<uie::simple_command_menu_node> m_nodes;
 };
 
+class AppendNewlineMenuNode : public uie::simple_command_menu_node {
+public:
+    AppendNewlineMenuNode(service_ptr_t<ConsoleWindow> window)
+        : simple_command_menu_node("Append newline", "", window->get_append_newline() ? uih::Menu::flag_checked : 0,
+            [window] { window->set_append_newline(!window->get_append_newline()); })
+    {
+    }
+};
+
 void ConsoleWindow::get_menu_items(uie::menu_hook_t& p_hook)
 {
     p_hook.add_node(new EdgeStyleMenuNode(this));
+    p_hook.add_node(new AppendNewlineMenuNode(this));
```

Comment:
I think we can get rid of the class by doing something like this:

https://github.com/reupen/columns_ui/blob/f0ca0bebf96333b71c3586437acf5be8bbc62803/foo_ui_columns/artwork.cpp#L67-L71

Reply:
I started off creating the node inline and didn't like it but OK.

Reply:
...unless you're trying to put emphasis on bumping up refcount of the parent object so that the menu action can be safely executed one last time.